students/KevinWaldron/lesson09/mailroom.py

```diff
@@ -0,0 +1,197 @@
+        :returns: formatted thank you string
+        """
+        return f"Dear {self.name},\nThank you for your very generous donation of ${donation:.2f}.  It \nwill go very far in supporting the Human Fund, \"Money for \nPeople.\"\n{'Sincerely':>40}\n{'Art Vandelay':>50}"
+
+    def __lt__(self, other):
+        return self.total_donations < other.total_donations
+
+    def __eq__(self, other):
+        return self.name == other.name and self._donations == other._donations
+
+class Donors:
+    """
+    A collection of donors
+    """
+    def __init__(self, donors=None):
+        if donors is None:
+            self._donors = []
+        else:
+            self._donors = donors
+
+    def add_donor(self, donor):
+        """
+        Add a new donor to the donors list
+        :param donor: the new donor to add
+        """
+        self._donors.append(donor)
+
+    def list_donors(self):
+        """ Creates a list of the donors by name """
+        return f"\n{'-'*10} Donors {'-'*10}\n" + '\n'.join([donor.name for donor in self._donors])
+
+    def print_donors(self):
+        """ Prints the list of donors """
+        print(self.list_donors())
+
+    def find_donor(self, name):
+        """
+        Find a donor by name
+        :param name: the name of the donor to search for
+        :return: The donor if found else None
+        """
+        for donor in self._donors:
+            if donor.name == name:
+                return donor
+        return None
+
+    def add_donation(self, name, amount):
+        """
+        Add a donation.
+        :param name: the name of the donor.
+        :param amount: the amount of the donation.the
+        :returns: the list of donations for the donor
+        """
+        donor = self.find_donor(name)
+        if donor is None:
+            donor = Donor(name)
+            self.add_donor(donor)
+        amount = float(amount)
+        donor.add_donation(amount)
+        print("\n" + donor.create_thank_you(amount))
+        return donor
+
+    def create_report(self):
+        """
+        Handles the create report menu selection
+        :returns: returns the report string
+        """
+        header = "\n{:<20}| Total Given | Num Gifts | Average Gift\n".format("Donor Name")
+        result = header
+        result += f"{'-' * (len(header) - 1)}"
+        donors_list = sorted(self._donors, reverse=True)
+        for donor in donors_list:
+            result += "\n{:<20}| ${:>10.2f} | {:>9} | ${:>11.2f}".format(donor.name, donor.total_donations, donor.num_donations, donor.avg_donation)
+        return result
+
+    def print_report(self):
+        """ Prints the report """
+        print(self.create_report())
+
+    def mail_everyone(self):
+        """ Handles the mail to everyone menu selection """
+        for donor in self._donors:
+            with open(donor.name.lower().replace(' ', "_") + ".txt", 'w') as f:
+                f.write(donor.create_thank_you(donor.total_donations))
+
```

Comment:
the classes look really great! 

minor nitpick - we technically should have print() functionality outside of data logic classes; consider use case where next feature is to have a UI instead of working with a console script. Now the print functionality is obsolete, but you have them nicely separated into separate functions so not a big deal.